applications/nrf5340_audio/src/modules/sd_card.h

```diff
@@ -8,55 +8,99 @@
+ *					The actual written size will be returned.
  *
- * @return	0 on success.
- *              -ENODEV SD init failed. SD card likely not inserted
- *		Otherwise, error from underlying drivers
+ * @retval	0 on success.
+ * @retval	-ENODEV SD init failed. SD card likely not inserted.
+ * @retval	Otherwise, error from underlying drivers.
  */
-int sd_card_write(char const *const filename, char const *const data, size_t *size);
+int sd_card_open_write_close(char const *const filename, char const *const data, size_t *size);
 
-/**@brief Read data from file into the buffer
+/**@brief Read data from file into the buffer.
  *
- * @param[in] filename	Name of the target file for reading, the default location is the
- *			root directoy of SD card, accept absolute path under root of SD card
- * @param[in] data	The buffer which will be filled by read file contents
- * @param[in,out] size	Pointer to the number of bytes which wait to be read from the file
- *			The actual read size will be returned
- *			If the actual read size is 0, there will be a warning message which
- *			indicates the file is empty
- * @return	0 on success.
- *              -ENODEV SD init failed. SD card likely not inserted
- *		Otherwise, error from underlying drivers
+ * @param[in]		filename	Name of the target file for reading, the default location is
+ *					the root directoy of SD card, accept absolute path under
+ *					root of SD card.
+ * @param[out]		buf		Pointer to the buffer to write the read data into.
+ * @param[in, out]	size		Pointer to the number of bytes which wait to be read from
+ *					the file. The actual read size will be returned. If the
+ *					actual read size is 0, there will be a warning message which
+					indicates that the file is empty.
+ *
+ * @retval	0 on success.
+ * @retval	-ENODEV SD init failed. SD card likely not inserted.
+ * @retval	Otherwise, error from underlying drivers.
+ */
+int sd_card_open_read_close(char const *const filename, char *const data, size_t *size);
+
+/**@brief	Open file on SD card.
+ *
+ * @param[in]	filename	Name of file to open.
+ *
+ * @retval	0 on success.
+ * @retval	-ENODEV SD init failed. SD likely not inserted.
+ * @retval	-EPERM SD card read is ongoing somewhere else.
+ * @retval	Otherwise, error from underlying drivers.
+ */
+int sd_card_open(char const *const filename, struct fs_file_t *f_seg_read_entry);
+
+/**@brief	Read segment on the open file on the SD card.
+ *
+ * @param[out]		buf	Pointer to the buffer to write the read data into.
+ * @param[in, out]	size	Number of bytes to be read from file.
+ *				The actual read size will be returned.
+ *				If the actual read size is 0, there will be a warning message.
+ *				which indicates that the file is empty.
+ *
+ * @retval	0 on success.
+ * @retval	-ENODEV SD init failed. SD likely not inserted.
+ * @retval	Otherwise, error from underlying drivers.
+ */
+int sd_card_read(char *const data, size_t *size, struct fs_file_t *f_seg_read_entry);
```

Comment:
since the data is being read into `data`, suggest removing the const, although this will work.